<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [3]:
# Membuat dataframes dari dictionary
import pandas as pd
data = pd.DataFrame({'Nama_Siswa':['Ani','Budi','Cici','Deni','Eko','Fahrul'],
                     'Fisika':[90, 80, 70, 85, 95, 100],
                     'Matematika':[80, 90, 85, 70, 85, 90],
                     'Biologi':[90, 85, 95, 100, 90, 85]},
                    index=[1,2,3,4,5,6])

data['Rata-Rata'] = data.mean(axis=1).round(2)
data

Nama_Siswa  Fisika  Matematika  Biologi  Rata-Rata
1        Ani      90          80       90      86.67
2       Budi      80          90       85      85.00
3       Cici      70          85       95      83.33
4       Deni      85          70      100      85.00
5        Eko      95          85       90      90.00
6     Fahrul     100          90       85      91.67

<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [14]:
from google.colab import drive
drive.mount('/content/gdrive')

Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).


In [15]:
# Membaca data dan memasukannya kedalam suatu variabel
diamond = pd.read_csv('/content/gdrive/My Drive/diamonds-1.csv')
diamond

Unnamed: 0  carat        cut color  ... price     x     y     z
0               1   0.23      Ideal     E  ...   326  3.95  3.98  2.43
1               2   0.21    Premium     E  ...   326  3.89  3.84  2.31
2               3   0.23       Good     E  ...   327  4.05  4.07  2.31
3               4   0.29    Premium     I  ...   334  4.20  4.23  2.63
4               5   0.31       Good     J  ...   335  4.34  4.35  2.75
...           ...    ...        ...   ...  ...   ...   ...   ...   ...
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[53940 rows x 11 columns]

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab:

1. a. Perbedaan antara list dengan series yaitu, sumbu pada series dapat diubah menjadi bukan numerik seperti a,b,c,d dan untuk memanggil data dari series dapat dilakukan dengan memanggil index/sumbu tersebut. Index pada list tidak dapat diubah. 

 b. Perbedaan antara series dengan dictionary yaitu, key/indeks pada series dapat diduplikat sedangkan pada dictionary tidak bisa. Selain itu, slicing dapat dilakukan pada series sedangkan pada dictionary tidak dapat dilakukan.



2. Perbedaan antara dataframe dengan series yaitu, dataframe merupakan objek dua dimensi yang terdiri dari index baris dan index kolom. Sedangkan series merupakan objek satu dimensi. Series tipe datanya harus seragam sedangkan dataframe dapat berbeda antar kolom. 

In [30]:
# code here
x = pd.Series(['Apel','Jeruk','Mangga','Jambu','Anggur','Leci','Melon'], index=['a','b','c','d','e','f','g'])
print(x)

a      Apel
b     Jeruk
c    Mangga
d     Jambu
e    Anggur
f      Leci
g     Melon
dtype: object


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [21]:
# code here

print("Data Pertama")
data_pertama = diamond.head()
print(data_pertama)

print('\n')

print("Data Terakhir")
data_terakhir = diamond.tail()
print(data_terakhir)

print('Baris pertama kolom price: ',diamond['price'][0])
print('Baris terakhir kolom color: ',diamond['color'][53939])

Data Pertama
   Unnamed: 0  carat      cut color clarity  ...  table  price     x     y     z
0           1   0.23    Ideal     E     SI2  ...   55.0    326  3.95  3.98  2.43
1           2   0.21  Premium     E     SI1  ...   61.0    326  3.89  3.84  2.31
2           3   0.23     Good     E     VS1  ...   65.0    327  4.05  4.07  2.31
3           4   0.29  Premium     I     VS2  ...   58.0    334  4.20  4.23  2.63
4           5   0.31     Good     J     SI2  ...   58.0    335  4.34  4.35  2.75

[5 rows x 11 columns]


Data Terakhir
       Unnamed: 0  carat        cut color  ... price     x     y     z
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[5 rows x 11 columns]
Baris pertama

<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!


Jawab: 

Deskriptif statistik adalah metode yang berhubungan dengan pengumpulan dan penyajian data sampai memberikan informasi yang berguna. Bentuk penyajian data dapat berupa grafis seperti membuat histogram, pie chart maupun numeris seperti mean, modus, median, standar deviasi dan lain-lain.

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [23]:
# code here
diamond.describe()

Unnamed: 0         carat  ...             y             z
count  53940.000000  53940.000000  ...  53940.000000  53940.000000
mean   26970.500000      0.797940  ...      5.734526      3.538734
std    15571.281097      0.474011  ...      1.142135      0.705699
min        1.000000      0.200000  ...      0.000000      0.000000
25%    13485.750000      0.400000  ...      4.720000      2.910000
50%    26970.500000      0.700000  ...      5.710000      3.530000
75%    40455.250000      1.040000  ...      6.540000      4.040000
max    53940.000000      5.010000  ...     58.900000     31.800000

[8 rows x 8 columns]

Jawab:

Rata-rata kolom price = 3932.799722 	
Standar deviasi kolom depth = 1.432621 	
Nilai maksimum kolom carat = 5.010000 	